File: problems/hard/morgan_and_string/__tests__/TestMorganString.java
package problems.hard.morgan_and_string.__tests__;
import problems.hard.morgan_and_string.MorganString;

public class TestMorganString {
    public static void main(String[] args) {
        if (!MorganString.morganAndString("ACA", "BCF").equals("ABCACF")) {
            throw new AssertionError("Test 1 failed");
        }
        if (!MorganString.morganAndString("JACK", "DANIEL").equals("DAJACKNIEL")) {
            throw new AssertionError("Test 2 failed");
        }
        System.out.println("✅ All Java tests passed!");
    }
}

File: problems/hard/morgan_and_string/__tests__/test_morgan_string.py
import sys
import os

sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from morgan_and_string import morganAndString

def test_case_1():
    assert morganAndString("ACA", "BCF") == "ABCACF"

def test_case_2():
    assert morganAndString("JACK", "DANIEL") == "DAJACKNIEL"
